### PR TITLE
"Update Block" button should probably specify that you only need to use it to update block metadata

### LINKED ISSUE #709: Typo on the Programs page when entering as a Guest
There's a typo on the Programs page when entering as a Guest.

The phrase "and well handle" should be "and we'll handle".

1. Go to 'http://seattle.civiform.com/loginForm'
2. Click on 'Continue'
3. The text with the typo is on the right

### LINKED ISSUE #842: "Update Block" button should probably specify that you only need to use it to update block metadata
**Describe the bug**
"Update Block" button should probably specify that you only need to use it to update block metadata (name and description) and not save what questions you have added/removed from the block.  Ideally, it's in a disabled state unless you have created a new block or changed the metadata.

### PR DESCRIPTION
**Pull request by thomasphung**
### Description
* In admin view, "Update Block" button is renamed to "Update Metadata" for specificity.
  * This button is now disabled by default until block metadata is changed. If form elements are empty, button is also disabled.
* Changed delete button background color to red.
* Fixes to i18n messages.

Peek 2021-04-30 18-10

### Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### Issue(s)
Fixes #842, fixes #709

### CODE REVIEW

File: universal-application-tool-0.0.1/app/assets/javascripts/main.ts [hunk 1]
```diff
@@ -40,6 +40,12 @@ function maybeHideElement(e: Event, id: string, parentId: string) {
   }
 }
 
+/** In admin program block edit form - enabling submit button when form is changed */
+function enableButton(event: Event) {
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/app/assets/javascripts/main.ts:44]: (blocking) the name `enableButton` seems very generic, but this enables a very specific button. Please make the functionality match the generality of the name OR make the name match the specificity of the functionality.

File: universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java [hunk 2]
```diff
@@ -216,29 +216,39 @@ private ContainerTag blockEditPanel(
...
       blockInfoForm.with(
           submitButton("Delete Block")
               .withId("delete-block-button")
               .attr(Attr.FORM, DELETE_BLOCK_FORM_ID)
-              .withClasses(Styles.MX_4, Styles.MY_1, Styles.INLINE));
+              .withClasses(
+                  Styles.MX_4,
+                  Styles.MY_1,
+                  Styles.BG_RED_500,
+                  "hover:" + Styles.BG_RED_700,
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java:250]: (blocking) there is a `StyleUtils#hover` method for this.

**Review by kendrickt — CHANGES_REQUESTED** [review]
Sweet preview of the feature! 

Looks like browser tests need to be fixed.

**Comment by kendrickt** [comment]
(blocking) test strings need to be updated, in both unit (error messages) and browser tests (update button id).

File: universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java (hunk 2 above)
Comment (thomasphung) [reply universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java]: Resolved

File: universal-application-tool-0.0.1/app/assets/javascripts/main.ts (hunk 1 above)
Comment (thomasphung) [reply universal-application-tool-0.0.1/app/assets/javascripts/main.ts]: Resolved, renamed function for simplicity.

**Review by kendrickt — APPROVED** [review]
looks good to me

**Review by kendrickt — CHANGES_REQUESTED** [review]
(blocking) It doesn't look like the button is grey-ed out any more. And it still changes color when it is disabled.

File: universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java [hunk 3]
```diff
@@ -216,29 +216,39 @@ private ContainerTag blockEditPanel(
...
+                .getContainer(),
+            FieldWithLabel.textArea()
+                .setId("block-description-textarea")
+                .setFieldName("description")
+                .setLabelText("Block description")
+                .setValue(blockForm.getDescription())
+                .getContainer(),
+            submitButton("Update Metadata")
+                .withId("update-block-button")
+                .withClasses(Styles.MX_4, Styles.MY_1, Styles.INLINE)
```
Comment (AWBaumann) [inline universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java:236]: Add a DISABLED type in StyleUtils and then this would be:

`.withClasses(Styles.MX_4, Styles.MY_1, Styles.INLINE, Styles.OPACITY_100, StyleUtils.disabled(Style.OPACITY_50)`

File: universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java [hunk 4]
```diff
@@ -216,29 +216,39 @@ private ContainerTag blockEditPanel(
...
+                .attr("disabled", ""));
```
[the remark this answers is not in the document]
Comment (thomasphung) [reply universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java:237]: Tried to do that but it only works when you load the page with nothing in the form fields (the HTML is generated server-side). The web page already throws a toast when the user tries to submit an empty form, but I could probably update the JS script to disable the button on empty forms.

File: universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java (hunk 3 above)
Comment (thomasphung) [reply universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java]: Resolved

File: universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (thomasphung) [reply universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java:242]: (blocking) Issues with Tailwind not changing the opacity of the button when disabled

[the remark this answers is not in the document]
Comment (thomasphung) [reply universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java:242]: Resolved

File: universal-application-tool-0.0.1/conf/messages [hunk 5]
```diff
@@ -21,7 +21,7 @@ button.nextBlock=Save and continue
 button.deleteEntity=Delete {0}
 
 # A toast message that displays when a program is not fully localized to the applicant's preferred locale.
-toast.localeNotSupported=We're sorry, this program is not fully translated to your preferred language.
+toast.localeNotSupported=We''re sorry, this program is not fully translated to your preferred language.
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/conf/messages:24]: (blocking) please revert changes to messages. It is outside the scope of this PR, and if we need them they can be submitted in a different PR.

File: universal-application-tool-0.0.1/test/support/ResourceCreator.java [hunk 6]
```diff
@@ -17,6 +17,7 @@
 import services.question.types.QuestionDefinition;
 import services.question.types.TextQuestionDefinition;
 
+/** Creates services that interface with the default database server for testing. */
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/test/support/ResourceCreator.java:20]: (blocking) let's keep this PR scope tight. You're welcome to submit a different PR to add comments and fix formatting.

File: universal-application-tool-0.0.1/test/support/ProgramBuilder.java [hunk 7]
```diff
@@ -244,6 +244,7 @@ public BlockBuilder withQuestionDefinitions(ImmutableList<QuestionDefinition> qu
       blockDefBuilder.setProgramQuestionDefinitions(pqds);
       return this;
     }
+
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/test/support/ProgramBuilder.java:247]: (blocking) let's keep this PR scope tight. You're welcome to submit a different PR to add comments and fix formatting.

**Review by kendrickt — CHANGES_REQUESTED** [review]

**Review by kendrickt — APPROVED** [review]
